### PR TITLE
Convert physics_common.F90 and physics_mg.F90 to C++

### PR DESCRIPTION
**Pull request by nelsonag**
This PR does two things:
1. Converts physics_common.F90 and physics_mg.F90 to C++, and
2. Removes the '_c' in function names within math_functions.h if the functions no longer exist in both Fortran and C++ namespaces.  This isn't a feature-add, but it is a slight entropy reducer.

This is pretty straight-forward, I guess its worthwhile to note that to do the above I had to move the `Materials % fissionable` attribute from the Fortran side to the C++ side.  This comes with the appropriate getters and setters, of course.

### CODE REVIEW

File: src/physics_mg.cpp [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+  // Begin banking the source neutrons
+  // First, if our bank is full then don't continue
+  if ((nu == 0) || (size_bank == bank_array_size)) return;
+
+  // Initialize the counter of delayed neutrons encountered for each delayed
+  // group.
+  double nu_d[MAX_DELAYED_GROUPS] = {0.};
+
+  p->fission = true;
+  // TODO: +1 for start?
```
Comment (paulromano) [inline src/physics_mg.cpp]: What's this about?

File: src/tracking.F90 [hunk 2]
```diff
@@ -17,7 +17,7 @@ module tracking
   use nuclide_header
   use particle_header
   use physics,            only: collision
-  use physics_mg,         only: collision_mg
+  ! use physics_mg,         only: collision_mg
```
Comment (paulromano) [inline src/tracking.F90]: Remove line

File: src/physics_mg.cpp (hunk 1 above)
Comment (nelsonag) [reply src/physics_mg.cpp]: Ah a note I left for myself that i no longer need.  Whoops.

**Comment by nelsonag** [comment]
@paulromano Thanks for the quick review; yes I am also working on physics.F90 as we speak.  I may end up submitting that in a few pieces: (1) low hanging fruit, (2) a larger move of more pieces of nuclide and materials to C++, and (3) the rest of physics.F90.  Right now i'm chugging through part (1), we will see if (2) and (3) are together or not a bit later.

**Pull request merged**